inc/classes/Buffer/class-cache.php

```diff
@@ -334,6 +328,35 @@ public function maybe_process_buffer( $buffer ) {
 		return $buffer . $footprint;
 	}
 
+	/**
+	 * Writes the cache file(s)
+	 *
+	 * @since 3.5
+	 * @author Remy Perona
+	 *
+	 * @param string $cache_filepath Absolute path to the cache file.
+	 * @param string $content Content to write in the cache file.
+	 * @return void
+	 */
+	private function write_cache_file( $cache_filepath, $content ) {
+		$gzip_filepath      = $cache_filepath . '_gzip';
+		$temp_filepath      = $cache_filepath . '_temp';
+		$temp_gzip_filepath = $gzip_filepath . '_temp';
+
+		if ( rocket_direct_filesystem()->exists( $temp_filepath ) ) {
+			return;
+		}
+
+		// Save the cache file.
+		rocket_put_content( $temp_filepath, $content );
+		rocket_direct_filesystem()->move( $temp_filepath, $cache_filepath );
+
+		if ( function_exists( 'gzencode' ) ) {
+			rocket_put_content( $temp_gzip_filepath, gzencode( $content, apply_filters( 'rocket_gzencode_level_compression', 3 ) ) );
```

Comment:
Missing a DocBlock here for the filter.

Reply:
I added it but I didn't add `@since` and `@author` because it's so old I have no idea

Reply:
Sure 👍